### PR TITLE
gnmi-tools - support for NSO flavor API adapter

### PR DESCRIPTION
**Pull request by micnovak**

### CODE REVIEW

File: docs/ConfD_gNMI_adapter.adoc [hunk 1]
```diff
@@ -241,11 +241,11 @@ Server is started by running  `./src/confd_gnmi_server.py` python script.
 .output
 [source, shell]
 ----
-usage: confd_gnmi_server.py [-h] [-v] [-t {api,demo}]
-                            [--logging {error,warning,info,debug}] [--insecure]
-                            [--port PORT] [--host HOST]
+usage: confd_gnmi_server.py [-h] [-v] [-t {api,demo,nso}]
```
Comment (jomiklos) [inline docs/ConfD_gNMI_adapter.adoc:244]: `api` vs `nso` -> maybe rename `api` to `confd` not to confuse with product name vs no product name and the difference?

File: docs/ConfD_gNMI_adapter.adoc [hunk 2]
```diff
@@ -241,11 +241,11 @@ Server is started by running  `./src/confd_gnmi_server.py` python script.
...
+                            [--logging {error,warning,info,debug}]
+                            [--insecure] [--port PORT] [--host HOST]
                             [-d {trace,debug,silent,proto}]
-                            [--confd-addr CONFD_ADDR] [--confd-port CONFD_PORT]
+                            [--confd-addr CONFD_ADDR] [--api-port API_PORT]
```
Comment (jomiklos) [inline docs/ConfD_gNMI_adapter.adoc:248]: "api" feels a bit too abstract - what "api" is this? is it confd-lib's maapi? if so, i'd use "maapi" to make it easier to understand the global picture...

File: src/confd_gnmi_api_adapter.py [hunk 3]
```diff
@@ -22,57 +25,57 @@
...
-        nslist = _confd.get_nslist()
+        nslist = _tm.get_nslist()
         self.module_to_pfx = {nsentry[-1]: nsentry[1] for nsentry in nslist}
         self.pfx_to_module = {nsentry[1]: nsentry[-1] for nsentry in nslist}
         self.ns_to_module = {nsentry[0]: nsentry[-1] for nsentry in nslist}
 
     # call only once!
     @staticmethod
-    def set_confd_debug_level(level):
+    def set_debug_level(level):
```
Comment (jomiklos) [inline src/confd_gnmi_api_adapter.py:56]: `set_maapi_debug_level()` not to confuse with global logging?

File: tests/test_common.py [hunk 4]
```diff
@@ -108,8 +107,8 @@ def test_client_argparse():
 def test_server_argparse():
     check_in_args(server_parse_args([]),
                   {"type": "demo", "logging": "warning", "confd_debug": "debug",
-                   "confd_addr": ApiAdapterDefaults.CONFD_ADDR,
-                   "confd_port": ApiAdapterDefaults.CONFD_PORT,
+                   "confd_addr": ApiAdapterDefaults.ADDR,
```
Comment (jomiklos) [inline tests/test_common.py:110]: `confd_addr` stays without change while port is renamed?

File: tests/test_common.py [hunk 5]
```diff
@@ -119,9 +118,9 @@ def test_server_argparse():
              {"dest": "confd_debug", "args": ["-d", "--confd-debug"],
```
Comment (jomiklos) [inline tests/test_common.py:118]: `--confd-debug` stays without rename?

**Review by jomiklos — APPROVED** [review]

**Review by martin-volf — APPROVED** [review]
Appreciate the code-style fixes; technically, they should be in a separate commit, but I'm not complaining.

**Pull request merged**